templates/app.py

```diff
@@ -0,0 +1,20 @@
+from new_metric_script import NewMetric
+
+import gradio as gr
+
+metric = NewMetric()
+metric_features = list(metric.features.keys())
+
+def compute(data):
+    metric_input = dict([(feat, data[feat]) for feat in metric_features])
+    return metric.compute(**metric_input)
+    
+iface = gr.Interface(
+  fn=compute, 
+  inputs=gr.inputs.Dataframe(headers=metric_features, col_width=len(metric_features), datatype="number"),
```

Comment:
Should the datatype also be specified programatically?

Reply:
Yes I was thinking about that, too. With #15 we should be able to infer that. I think we also need to add a case for when we can't infer the type (e.g. somebody implements a metric for a new modality). Maybe we can then just display a text saying that the widget is not available for that metric but it could be implemented in `app.py`.